app/charges/models.py

```diff
@@ -20,16 +20,27 @@
 ## This is used for the progress_notes column
 ##
 class MutableList(Mutable, list):
+    def findIndex(self, id):
+        index = 0
+        correctIndex = 0
+        for note in self: # find the index of the given note_id
+            if int(note[2]) == id:
+                correctIndex = index
+            else: index += 1
+        return correctIndex
```

Comment:
If this doesn't find a matching id in the array it returns an index of 0 which will delete the first note. I think there should be a check to make sure a match was found.